From 2605a8fb0cb5ba288572a7cd791e0f4da0d11c65 Mon Sep 17 00:00:00 2001
From: Rowell Cruz <rcruz@blackberry.com>
Date: Wed, 18 Sep 2013 16:35:09 -0400
Subject: [PATCH 32/61] Fix issue in Contacts involving saving and retrieving
 notes

Reviewed by Bryan Higgins <bhiggins@blackberry.com>
Tested by Daniel Audino <daudino@blackberry.com>
---
 .../src/blackberry10/native/pim_contacts_qt.cpp         |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/plugin/org.apache.cordova.blackberry10.pimlib/src/blackberry10/native/pim_contacts_qt.cpp b/plugin/org.apache.cordova.blackberry10.pimlib/src/blackberry10/native/pim_contacts_qt.cpp
index 6bf6609..532d293 100644
--- a/plugin/org.apache.cordova.blackberry10.pimlib/src/blackberry10/native/pim_contacts_qt.cpp
+++ b/plugin/org.apache.cordova.blackberry10.pimlib/src/blackberry10/native/pim_contacts_qt.cpp
@@ -783,7 +783,13 @@ void PimContactsQt::populateField(const bbpim::Contact& contact, bbpim::Attribut
         Json::Value val;
         SubKindToStringMap::const_iterator typeIter = _subKindAttributeMap.find(currentAttr.subKind());
 
-        if (typeIter != _subKindAttributeMap.end()) {
+        // Check note type first
+        if (kind == bbpim::AttributeKind::Note) {
+            std::string note = currentAttr.value().toStdString();
+            note = replaceString(note);
+            contactItem["note"] = Json::Value(note);
+            break;
+        } else if (typeIter != _subKindAttributeMap.end()) {
             if (isContactField) {
                 val["type"] = Json::Value(typeIter->second);
 
@@ -810,11 +816,6 @@ void PimContactsQt::populateField(const bbpim::Contact& contact, bbpim::Attribut
                     }
                 }
             }
-        } else if (kind == bbpim::AttributeKind::Note) {
-            std::string note = currentAttr.value().toStdString();
-            note = replaceString(note);
-            contactItem["note"] = Json::Value(note);
-            break;
         }
     }
 }
@@ -1544,7 +1545,7 @@ void PimContactsQt::createAttributeSubKindMap()
     _attributeSubKindMap["YahooMessegerJapan"] = bbpim::AttributeSubKind::InstantMessagingYahooMessengerJapan;
     _attributeSubKindMap["BbPlaybook"] = bbpim::AttributeSubKind::VideoChatBbPlaybook;
     _attributeSubKindMap["ringtone"] = bbpim::AttributeSubKind::SoundRingtone;
-    _attributeSubKindMap["note"] = bbpim::AttributeSubKind::Invalid;
+    _attributeSubKindMap["note"] = bbpim::AttributeSubKind::Other;
 }
 
 void PimContactsQt::createKindAttributeMap() {
-- 
1.7.10.msysgit.1

